### ISSUE TITLE
I need to limit triangle sharpness to 2 samples, what's the equation?

### ISSUE THREAD

**Issue by elanhickler** [RobinSchmidt/RS-MET#105]
a triangle rises for  1/frequency/2 * 44100 samples

with a skew parameter, we can make one side of the triangle rise faster than the other, but we never want the triangle to rise faster than 2 samples per cycle,

at 1/4*samplerate frequency, the triangle has 4 samples per cycle, 2 samples for one slope, multiplied by the slope factor which is 0 to 1, and so at that frequency, our slope must be fully unskewed, so our slope is 1. So, between 0hz and 11025hz, our slope must be limited from 0 to 1.

f = 11025
sr = 44100
0 < x < 1

1 = clip(1/f/2 * sr * x, 0, 1)
x == 1

how do we solve for x given any frequency and sample rate? x is our slope maximum sharpness to achieve a minimum of 2 samples per half cycle

**Comment by RobinSchmidt** [comment]
i'm a bit confused. are you talking about the tri-saw waveform and want an equation for the limit of the skew-factor (between -1..+1, where 0 means triangle) as function of frequency, such that at a freq of fs/2, the skew-limit is 0?

**Comment by elanhickler** [comment]
sure, i was using 0 to 1 instead of-1 to +1, but yes.

**Comment by elanhickler** [comment]
no longer relevant, this will be solved in your recent antialiasing work

**Comment by RobinSchmidt** [comment]
yes - triangle anti-aliasing is the prototypical target case for blamps
